src/collections/_documentation/platforms/javascript/angular.md

```diff
@@ -16,23 +16,81 @@ yarn add @sentry/browser
 npm install @sentry/browser
 ```
 
+Then initialize new Sentry instance and configure Angular with `ErrorHandler` provided and explained below:
+
 ```typescript
 import { BrowserModule } from "@angular/platform-browser";
 import { NgModule, ErrorHandler, Injectable } from "@angular/core";
+import { HttpErrorResponse } from "@angular/common/http";
 
+import { environment } from "../environments/environment";
 import { AppComponent } from "./app.component";
 
 import * as Sentry from "@sentry/browser";
 
 Sentry.init({
-  dsn: "___PUBLIC_DSN___"
+  dsn: "___PUBLIC_DSN___",
+  // TryCatch has to be configured to disable XMLHttpRequest wrapping, as we are going to handle
+  // http module exceptions manually in Angular's ErrorHandler and we don't want it to capture same error twice.
+  // Please note that TryCatch configuration requires at least @sentry/browser v5.16.0.
+  integrations: [new Sentry.Integrations.TryCatch({
+    XMLHttpRequest: false,
+  })],
 });
 
 @Injectable()
 export class SentryErrorHandler implements ErrorHandler {
   constructor() {}
+
+  extractError(error) {
+    // Try to unwrap zone.js error.
+    // https://github.com/angular/angular/blob/master/packages/core/src/util/errors.ts
+    if (error && error.ngOriginalError) {
+      error = error.ngOriginalError;
+    }
+
+    // We can handle messages and Error objects directly.
+    if (typeof error === "string" || error instanceof Error) {
+      return error;
+    }
+
+    // If it's http module error, extract as much information from it as we can.
+    if (error instanceof HttpErrorResponse) {
+      // `error` property of http exception can be either `Error` which we can use.
+      if (error.error instanceof Error) {
+        return error.error;
+      }
+
+      // `ErrorEvent` which can provide us with the message but no stack.
+      if (error.error instanceof ErrorEvent) {
+        return error.error.message;
+      }
+
+      // Or the request body itself, which we can use as a message instead.
+      if (typeof error.error === "string") {
+        return `Server returned code ${error.status} with body "${error.error}"`;
+      }
+
+      // If we don't have any detailed information, fallback to the request message itself.
+      return error.message;
+    }
+
+    // Skip if there's no error, and let user decide what to do with it.
+    return null;
+  }
+
   handleError(error) {
-    const eventId = Sentry.captureException(error.originalError || error);
+    let extractedError = this.extractError(error) || "Handled unknown error";
+
+    // Capture handled exception and send it to Sentry.
+    const eventId = Sentry.captureException(extractedError);
+    
+    // When in development mode, log the error to console for immediate feedback.
+    if (!environment.production) {
+      console.error(extractedError);
+    }
```

Comment:
If I'm reading this right, we're not actually skipping reporting, we're just reporting that we got something we didn't know how to deal with (`Handled unknown error`). In that case, can we attach the thing we did get as `extra`, for debugging purposes?

Could we do something like this? (GH won't let me put it as a suggestion, so for the purposes of reasonable syntax highlighting I've copied the whole class an elided parts I didn't change.)

```typescript
export class SentryErrorHandler implements ErrorHandler {
  constructor() {}

  extractError(error) {
    // ...

    // Nothing meaningful to extract from any of the standard spots
    // Handle this case in `handleError()`
    return null;
  }

  handleError(error) {
    // Attempt to extract an Error object or a message
    let extractedError = this.extractError(error);

    // Send error to Sentry
    let eventId;
    if (extractedError) {
      eventId = Sentry.captureException(extractedError);
    } 
    else {
      // If we weren't able to extract data from the usual spots, attach the
      // originally-captured object as `extra` for debugging purposes
      Sentry.withScope((scope) => {
        scope.setExtra("capturedError", JSON.stringify(error));
      });
      eventId = Sentry.captureException("Handled unknown error");
    }

    // When in development mode, log the error to console for immediate feedback.
    if (!environment.production) {
      console.error(
        extractedError || `Handled unknown error: ${JSON.stringify(error)}`
      );
    }

    // ...
  }
}
```

Reply:
Oh, this case happens only if `error` is `undefined`, which is pointless to provide in extras. I have no idea how this can happen, but it happened to me once 😅

Reply:
What if they end up with some custom error class which doesn't inherit from `Error`? IDK how Angular works - could something like that end up coming through this path, or no?

Reply:
I don't think so, Angular errors all inherit. If someone hacks it and provides something that we don't support (I dont believe it'll ever happen), then they just need to update their code. The docs should stay generic imo.

Reply:
👍🏻